File: glossier/exercise/ruby/lib/exercise/models/product.rb
module Exercise
  class Product
    attr_reader :category, :id, :name

    # category: String
    #   The category that the product belongs to, e.g.: skincare, makeup.
    # id: Integer
    #   The unique id of the product.
    # name: String
    #   The name of the product.
    def initialize(category:, id:, name:)
      @category, @id, @name = category, id, name
    end
  end
end

File: glossier/exercise/javascript/lib/exercise/models/product.js
var exports = module.exports = {}

/**
 * category: String
 *   The category that the product belongs to, e.g.: skincare, makeup.
 * id: Number
 *   The unique id of the product.
 * name: String
 *   The name of the product.
 */
exports.Product = class Product {
  constructor(category, id, name) {
    this.category = category
    this.id = id
    this.name = name
  }
}
